src/k8s/pkg/k8sd/api/cluster_delete.go

```diff
@@ -0,0 +1,33 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+	snaputil "github.com/canonical/k8s/pkg/snap/util"
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microcluster/state"
+)
+
+func (e *Endpoints) deleteCluster(s *state.State, r *http.Request) response.Response {
```

Comment:
As a generic comment on all these, please do not ignore the errors. But remember here we are working in cleanup mode. Finding an error does not necessarily mean fail early. Maybe you can keep going, then return a list of all errors that occured at the end with something like `errors.Join`